Add function to derive group info in trusted dealer setting (#132)
cfrg/draft-irtf-cfrg-frost: draft-irtf-cfrg-frost.md

    vss_verify(share_i, commitment)
      (i, sk_i) = share_i
      S_i = ScalarBaseMult(sk_i)
      S_i' = G.Identity()
      for j in range(0, t-1):
        S_i' += vss_commitment_j * i^j
      if S_i == S_i':
        return 1
      return 0
~~~

We now define how the coordinator and signing participants can derive group info, which is an input into the FROST signing protocol.

~~~
    derive_group_info(MAX_SIGNERS, THRESHOLD_LIMIT, vss_commitment):

    Inputs:
    - MAX_SIGNERS, the number of shares to generate, an integer
    - THRESHOLD_LIMIT, the threshold of the secret sharing scheme, an integer
    - vss_commitment: A VSS commitment to a secret polynomial f.

    Outputs:
    - PK, the public key representing the group
    - signer_public_keys, a list of MAX_SIGNERS public keys PK_i for i=1,...,n, where PK_i is the public key for participant i.

    derive_group_info(n, t, vss_commitment)
      PK = vss_commitment[0]
      signer_public_keys = []
      for i in range(1, n):
        PK_i = G.Identity()

>         for j in range(0, t-1):

Reviewer: ```suggestion
        for j in range(0, THRESHOLD_LIMIT-1):
```

      return 0
~~~

We now define how the coordinator and signing participants can derive group info, which is an input into the FROST signing protocol.

~~~

>     derive_group_info(n, t, vss_commitment):

Reviewer: ```suggestion
    derive_group_info(MAX_SIGNERS, THRESHOLD_LIMIT, vss_commitment):
```

We now use MAX_SIGNERS and THRESHOLD_LIMIT instead of n and t, respectively. (Side bar: maybe we should rename THRESHOLD_LIMIT to MIN_SIGNERS?)

Author: I would prefer keeping threshold in the name, to keep in line with existing terminology.

>     derive_group_info(n, t, vss_commitment):

Reviewer 2: This is the signing group? As opposed to the algebraic structure?

Author: We use group info earlier in the document as an input to each signer with these keys.

    Inputs:

>     - n, the number of shares to generate, an integer

Reviewer: ```suggestion
    - MAX_SIGNERS, the number of shares to generate, an integer
```

>     - t, the threshold of the secret sharing scheme, an integer

Reviewer: ```suggestion
    - THRESHOLD_LIMIT, the threshold of the secret sharing scheme, an integer
```

    - vss_commitment: A VSS commitment to a secret polynomial f.

    Outputs:
    - PK, the public key representing the group

>     - signer_public_keys, a set of n public keys PK_i for i=1,...,n, where PK_i is the public key for participant i.

Reviewer: ```suggestion
    - signer_public_keys, a list of MAX_SIGNERS public keys PK_i for i=1,...,n, where PK_i is the public key for participant i.
```

    derive_group_info(n, t, vss_commitment)
      PK = vss_commitment[0]

>       signer_public_keys = {}

Reviewer: ```suggestion
      signer_public_keys = []
```

      for i in range(1, n):

>         PK_i = SUM(commitment[0], commitment[t-1]){A_j}: A_j*(i^j)

Reviewer: We don't use this notation anymore -- can you please rewrite it using a loop?

Author: Addressed this and the loop in verify share.

>         signer_public_keys[i] = PK_i

Reviewer: ```suggestion
        signer_public_keys.append(PK_i)
```
